cli/config/configfile/file.go

```diff
@@ -254,10 +255,13 @@ func decodeAuth(authStr string) (string, string, error) {
 // GetCredentialsStore returns a new credentials store from the settings in the
 // configuration file
 func (configFile *ConfigFile) GetCredentialsStore(registryHostname string) credentials.Store {
+	var credsStore credentials.Store
 	if helper := getConfiguredCredentialStore(configFile, registryHostname); helper != "" {
-		return newNativeStore(configFile, helper)
+		credsStore = newNativeStore(configFile, helper)
+	} else {
+		credsStore = credentials.NewFileStore(configFile)
 	}
-	return credentials.NewFileStore(configFile)
+	return credentials.NewOAuthStore(credsStore, manager.NewManager())
```

Comment:
Hmm, I find it easier to reason about if we have a single point managing both. Otherwise, if someone does `GetCredentialsStore("some-other-registry").GetAll()` they're going to get the `access-token` and `refresh-token` specific keys that we shouldn't leak out/callers shouldn't concern themselves with. There might be other things that the wrapper does that should be abstracted away by the wrapper.